TU20Bot/Program.cs

```diff
@@ -25,16 +26,24 @@ internal class Program {
             handler = new Handler(client);
 
             await handler.init();
-            
+
             await client.LoginAsync(TokenType.Bot, token);
             await client.StartAsync();
 
             // Run server on another thread.
             new Thread(() => server.RunAsync().GetAwaiter().GetResult()).Start();
 
+            // Run email checker on another thread
+            new Thread(new ThreadStart(() => {
+                // Keep the thread running forever
+                do {
+                    new EmailChecker(config, client).checkForEmail();
+                } while (true);
+            })).Start();
```

Comment:
A thread is not nessesary for this... It will take up a fair bit of computing power, it perhaps would make sense if you had run the check every 1 minute or something (via Thread.Sleep)?

I think you can tie this to some sort of event instead of a forever running thread. For example
```
someUserEmailChangingFunction() {
  // change something to do with emails...
  EmailChecker.checkForEmail(); // run the check!!
}

// Maybe you need to run it every time something happens in discord, DiscordSocketClient has your back!
client.UserJoined += () => {
  EmailChecker.checkForEmails(); // run every time someone joins the server
}

client.MessageRecieved = () => {
  EmailChecker.checkForEmails(); // run every time someone sends a message
}

client.RoleAssigned = () => {
  EmailChecker.checkForEmails(); // run every time some gets assigned a role
}
```

P.S. if you use the discord methods you will usually have to take in some parameter (usually some kind of User/Message/Role/Guild object) and probably return `Task.CompletedTask`.

Reply:
That's a good idea. I'm thinking of just running it once every hour, instead of tying it to an event. Cuz in case that event doesn't occur for a certain period of time (say five or six hours), we will have a delay in assigning the roles.

Reply:
Its also cuz most of the discord events are based on human activity and I don't trust those inferior creatures known as humans. 👽

Reply:
:/ I don't know what requires something like this though. If an email can only be added every time someone sends a message or joins the server etc. you should run this check on that event. That way:
 - [x] super efficient
 - [x] no delay